Sources/SotoCore/AWSClient.swift

```diff
@@ -419,44 +426,47 @@ extension AWSClient {
                     configuration: serviceConfig
                 )
             },
-            execute: { request, eventLoop, logger in
-                return self.httpClient.execute(request: request, timeout: serviceConfig.timeout, on: eventLoop, logger: logger, stream: stream)
+            execute: { request, eventLoop, context in
+                return self.httpClient.execute(request: request, timeout: serviceConfig.timeout, on: eventLoop, context: context, stream: stream)
             },
             processResponse: { response in
                 return try self.validate(operation: operationName, response: response, serviceConfig: serviceConfig)
             },
             config: serviceConfig,
-            on: eventLoop,
-            logger: logger
+            context: context,
+            on: eventLoop
         )
     }
 
     /// internal version of execute
     internal func execute<Output>(
         operation operationName: String,
         createRequest: @escaping () throws -> AWSRequest,
-        execute: @escaping (AWSHTTPRequest, EventLoop, Logger) -> EventLoopFuture<AWSHTTPResponse>,
+        execute: @escaping (AWSHTTPRequest, EventLoop, Context) -> EventLoopFuture<AWSHTTPResponse>,
         processResponse: @escaping (AWSHTTPResponse) throws -> Output,
         config: AWSServiceConfig,
-        on eventLoop: EventLoop? = nil,
-        logger: Logger = AWSClient.loggingDisabled
+        context: Context,
+        on eventLoop: EventLoop? = nil
     ) -> EventLoopFuture<Output> {
         let eventLoop = eventLoop ?? eventLoopGroup.next()
-        let logger = logger.attachingRequestId(Self.globalRequestID.add(1), operation: operationName, service: config.service)
+        var context = context
+        context.baggage.awsService = config.service
+        context.baggage.awsOperation = operationName
+        context.baggage.awsRequestId = Self.globalRequestID.add(1)
```

Comment:
Have to make a copy of the context here to add values to the baggage

Reply:
That'd be slightly cleaner, we did discuss this here https://github.com/slashmo/gsoc-swift-baggage-context/pull/34#discussion_r488048198 although I can't find your response, in that situation my suggestion was to return a newly constructed `DefaultContext` which you rightly pointed out would cause confusion. But now that baggage is a `var` something of this form should be possible.
```
extension Context {
    func withBaggage(_: (inout Baggage) -> ()) -> Self
}
```